packages/firestore/src/local/shared_client_state.ts

```diff
@@ -399,7 +395,7 @@ export class WebStorageSharedClientState implements SharedClientState {
       `^${CLIENT_STATE_KEY_PREFIX}_${persistenceKey}_([^_]*)$`
     );
     this.mutationBatchKeyRe = new RegExp(
-      `^${MUTATION_BATCH_KEY_PREFIX}_${persistenceKey}_([^_]*)_(\\d*)$`
+      `^${MUTATION_BATCH_KEY_PREFIX}_${persistenceKey}_(\\d*)(?:_(.*))$`
```

Comment:
I'm trying to figure out two things:

1. Why `\\d*` instead of `\\d+`?
2. Why `(?:_(.*))$` instead of just `_(.*)$`?

Am I missing something or would it be more precise / simple to do:

    `^${MUTATION_BATCH_KEY_PREFIX}_${persistenceKey}_(\\d+)_(.*)$`

Reply:
Oh...  It looks like you only add _blah for authenticated users.  But then for this regex to work you'd need a `?` after the non-capturing group.  i.e. `...(?:_(.*))?$` right?  

Should there be a test for unauthenticated mutation batches?